### DISCUSSION TITLE
Search 'Any searchable Text' for Albums, Artists, Playlists, Radio Stations, Podcasts, Podcast Episodes and Videos

### DISCUSSION THREAD

**Discussion by DJ-SteffEN** [ampache/ampache#4016] [Q&A]
Hello,

I have extended the search functions -> see [Ajax in the header search](https://inthemix.at/).
Next, I wanted to create a rule for 'anywhere' for Albums, Artists, Playlists, Radio Stations, Podcasts, Podcast Episodes and Videos, just like for Songs.
Unfortunately, I am failing miserably with the SQL queries (Model\Search.php).
Does anyone here have a complete overview?

With kind regards 
DJ-Steffen

**Comment by lachlan-00** [comment]
do you mean you want to have 'anywhere' available for each search type?

and then you want them to return results in the header search bar?

The search bar runs each search from the ajax handler
`src/Application/Api/Ajax/Handler/SearchAjaxHandler.php`

To add anywhere into each option you'd just copy from the SongSearch code
`src/Module/Playlist/Search/SongSearch.php`

it's part of the anywhere case but it would need one for each type and editing to fit.
```
                case 'anywhere':
                    // 'anywhere' searches song title, song filename, song genre, album title, artist title, label title and song comment
```

Genre searching is also slow as hell and one reason why i didn't extend it to everything so avoid genre for the other types

**Comment by DJ-SteffEN** [comment]
Hello lachlan-00,

> do you mean you want to have 'anywhere' available for each search type?

Yes, I would like to have 'everywhere' for each type, like in the song search.

> and then you want them to return results in the header search bar?

No, I have already extended src/Application/Api/Ajax/Handler/SearchAjaxHandler.php.

> it's part of the anywhere case but it would need one for each type and editing to fit.

I wanted to do that, but I fail with the SQL queries and yes it is a lot of work.

With kind regards
DJ-Steffen

**Reply by lachlan-00** [reply]
okay, i will look at making a generic everywhere for all types. It won't take me very long as i started working on Ampache for search so i know it very well.
